src/data/array/table.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2018 Keeley Hoek. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Keeley Hoek
+-/
+
+import data.list
+import data.array.defs
+
+import tactic.lint
+
+/-!
+# Tables
+
+A `table` is a self-resizing array-backed list,
+which uses opaque references called `table_ref`s for access.
+-/
+
+universe variables u v w z
+
+attribute [inline] bool.decidable_eq option.is_some option.is_none list.head
```

Comment:
I think the canonical place to apply `attribute [inline]` is in `logic/basic`.

Reply:
I've tried to move things to the appropriate places.